### PR TITLE
fix: display shipping page only when user order new modem

### PR DESCRIPTION
**Pull request by Steffy29**
| Question         | Answer
| ---------------- | ---
| Branch?          | `master`
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | no
| Tickets          | Fix #NASD-6058
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~Breaking change is mentioned in relevant commits~

## Description

Display shipping for order modem only when it's necessary.

## Related
- [X] TRANS-50271

### CODE REVIEW

File: packages/manager/apps/telecom/src/app/telecom/pack/move/offers/move-offers.controller.js [hunk 1]
```diff
@@ -355,9 +368,44 @@ export default class PackMoveOffersCtrl {
...
+          optionLabel = this.$translate.instant(
+            `pack_move_modem_${option.name}`,
+            {
+              price: option.price ? option.price.text : '',
+            },
+          );
+        }
+        set(option, 'label', optionLabel);
+        return option;
+      });
```
Comment (anooparveti) [inline packages/manager/apps/telecom/src/app/telecom/pack/move/offers/move-offers.controller.js:399]: **nitpick:** `Array.prototype.map` returns a new array. But its return value is not used/captured. Please use appropriate array method.

File: packages/manager/apps/telecom/src/app/telecom/pack/migration/offers/pack-migration-offers.controller.js [hunk 2]
```diff
@@ -179,13 +184,42 @@ export default class TelecomPackMigrationOffersCtrl {
...
+          optionLabel = this.$translate.instant(
+            `telecom_pack_migration_modem_${option.name}`,
+            {
+              price: option.price ? option.price.text : '',
+            },
+          );
+        }
+        set(option, 'label', optionLabel);
+        return option;
+      });
```
Comment (anooparveti) [inline packages/manager/apps/telecom/src/app/telecom/pack/migration/offers/pack-migration-offers.controller.js:216]: **nitpick:** `Array.prototype.map` returns a new array. But its return value is not used/captured. Please use appropriate array method.

**Review by anooparveti — CHANGES_REQUESTED** [review]

File: packages/manager/apps/telecom/src/app/telecom/pack/move/offers/move-offers.controller.js (hunk 1 above)
Comment (Steffy29) [reply packages/manager/apps/telecom/src/app/telecom/pack/move/offers/move-offers.controller.js:399]: OK I understand your feedback, but what will you suggest for appropriate array method?

**Review by anooparveti — APPROVED** [review]

**Review by sachinrameshn — APPROVED** [review]

File: packages/manager/apps/telecom/src/app/telecom/pack/migration/confirm/pack-migration-confirm.html [hunk 3]
```diff
@@ -95,7 +95,7 @@ <h3 data-translate="telecom_pack_migration_doing"></h3>
                             >
                                 <th
                                     scope="row"
-                                    data-translate="{{'telecom_pack_migration_confirm_modem_rental_' + $ctrl.process.selectedOffer.modem}}"
+                                    data-translate="{{'telecom_pack_migration_confirm_modem_rental'}}"
```
Comment (ghyenne) [inline packages/manager/apps/telecom/src/app/telecom/pack/migration/confirm/pack-migration-confirm.html:98]:
```suggestion
                                    data-translate="telecom_pack_migration_confirm_modem_rental"
```

File: packages/manager/apps/telecom/src/app/telecom/pack/migration/offers/pack-migration-offers.controller.js [hunk 4]
```diff
@@ -179,11 +187,50 @@ export default class TelecomPackMigrationOffersCtrl {
...
+              },
+            );
+          } else if (!offer.modemRental) {
+            optionLabel = this.$translate.instant(
+              `telecom_pack_migration_modem_${option.name}`,
+              {
+                price: option.price ? option.price.text : '',
+              },
+            );
+          } else if (
```
Comment (ghyenne) [inline packages/manager/apps/telecom/src/app/telecom/pack/migration/offers/pack-migration-offers.controller.js:209]:
```suggestion
          if (offer.needNewModem || !offer.modemRental) {
            optionLabel = this.$translate.instant(
              `telecom_pack_migration_modem_${option.name}`,
              {
                price: option.price ? option.price.text : '',
              },
            );
          } else if (
```

File: packages/manager/apps/telecom/src/app/telecom/pack/move/offers/move-offers.controller.js [hunk 5]
```diff
@@ -355,9 +370,52 @@ export default class PackMoveOffersCtrl {
...
+            },
+          );
+        } else if (!offer.prices.modemRental.price) {
+          optionLabel = this.$translate.instant(
+            `pack_move_modem_${option.name}`,
+            {
+              price: option.price ? option.price.text : '',
+            },
+          );
+        } else if (
```
Comment (ghyenne) [inline packages/manager/apps/telecom/src/app/telecom/pack/move/offers/move-offers.controller.js:391]:
```suggestion
        if (offer.needNewModem || !offer.prices.modemRental.price) {
          optionLabel = this.$translate.instant(
            `pack_move_modem_${option.name}`,
            {
              price: option.price ? option.price.text : '',
            },
          );
        } else if (
```

File: packages/manager/apps/telecom/src/app/telecom/pack/move/resume/move-resume.html [hunk 6]
```diff
@@ -96,16 +96,16 @@ <h3 data-translate="pack_move_doing"></h3>
                                 ></td>
                             </tr>
                             <tr
-                                data-ng-if="$ctrl.offer.selected.offer.modemRental"
+                                data-ng-if="$ctrl.MODEM_LIST.includes($ctrl.offer.selected.offer.modem)"
                             >
                                 <th
                                     scope="row"
-                                    data-translate="{{'pack_move_confirm_modem_rental_' + $ctrl.offer.selected.offer.modem}}"
+                                    data-translate="{{'pack_move_confirm_modem_rental'}}"
```
Comment (ghyenne) [inline packages/manager/apps/telecom/src/app/telecom/pack/move/resume/move-resume.html:103]:
```suggestion
                                    data-translate="pack_move_confirm_modem_rental"
```

**Review by ghyenne — CHANGES_REQUESTED** [review]

**Review by ghyenne — APPROVED** [review]

**Review by anooparveti — DISMISSED** [review]

**Review by sachinrameshn — DISMISSED** [review]

**Review by anooparveti — APPROVED** [review]

**Review by ghyenne — APPROVED** [review]

**Review by sachinrameshn — APPROVED** [review]

**Review by sachinrameshn — DISMISSED** [review]

**Review by sachinrameshn — APPROVED** [review]

**Pull request merged**